apps/web/src/components/follow-button/follow-button.tsx

```diff
@@ -0,0 +1,62 @@
+'use client'
+
+import { Button } from '@/components/ui/button'
+import { Skeleton } from '@/components/ui/skeleton'
+import { useAuth } from '@/context/auth'
+import { useLanguage } from '@/context/language'
+import { useFollowers } from '@/hooks/use-followers'
+import { getFollowers } from '@/services/api/followers/get-followers'
+import { useQuery } from '@tanstack/react-query'
+
+type FollowButtonProps = { profileId: string }
+export const FollowButton = ({ profileId }: FollowButtonProps) => {
+  const { user } = useAuth()
+  const { dictionary } = useLanguage()
+  const { handleFollow, handleRemoveFollow } = useFollowers()
+
+  const { data: followers, isLoading } = useQuery({
+    queryKey: ['followers', profileId],
+    queryFn: async () => await getFollowers(profileId),
+  })
+
+  const userFollow = followers?.find((f) => f.follower_id === user?.id)
```

Comment:
Maybe rename userFollow to isFollower 

```typescript
const isFollower = followers?.find((f) => f.follower_id === user?.id)

Reply:
Thiis does not return a boolean. Returns the user's follow if there is one, and if so, the follow id is used to unfollow